azurerm/internal/services/containers/kubernetes_cluster_node_pool_resource_test.go

```diff
@@ -596,6 +597,27 @@ func testAccKubernetesClusterNodePool_spot(t *testing.T) {
 	})
 }
 
+func TestAccKubernetesClusterNodePool_upgradeSettings(t *testing.T) {
+	checkIfShouldRunTestsIndividually(t)
+	testAccKubernetesClusterNodePool_upgradeSettings(t)
+}
+
+func testAccKubernetesClusterNodePool_upgradeSettings(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_cluster_node_pool", "test")
+	r := KubernetesClusterNodePoolResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.upgradeSettingsConfig(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("upgrade_settings.0.max_surge").HasValue("2"),
+			),
+		},
+		data.ImportStep(),
```

Comment:
can we add tests to update this value and then remove it:

```suggestion
		{
 			Config: r.upgradeSettingsConfig(data, "2"),
 			Check: resource.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
 				check.That(data.ResourceName).Key("upgrade_settings.#").HasValue("1"),
 				check.That(data.ResourceName).Key("upgrade_settings.0.max_surge").HasValue("2"),
 			),
 		},
 		data.ImportStep(),
		{
 			Config: r.upgradeSettingsConfig(data, "4"),
 			Check: resource.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
 				check.That(data.ResourceName).Key("upgrade_settings.#").HasValue("1"),
 				check.That(data.ResourceName).Key("upgrade_settings.0.max_surge").HasValue("2"),
 			),
 		},
 		data.ImportStep(),
		{
 			Config: r.upgradeSettingsConfig(data, ""),
 			Check: resource.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
 				check.That(data.ResourceName).Key("upgrade_settings.#").HasValue("0"),
 			),
 		},
 		data.ImportStep(),
```